docs/ops/doc/IntrospectingOps.md

```diff
@@ -0,0 +1,124 @@
+
+```java
+System.out.println(Ops.signature(op));
+```
+
+might print out (again using `scijava-ops-image` version `1.0.0`):
+
+```
+|Reduction:|ParamsReduced:1|OriginalInfo:|Info:org.scijava.ops.image.filter.gauss.Gaussians.gaussRAISingleSigma(net.imglib2.RandomAccessibleInterval<I>,double,net.imglib2.outofbounds.OutOfBoundsFactory<I, net.imglib2.RandomAccessibleInterval<I>>,net.imglib2.RandomAccessibleInterval<O>)@1.0.0{}
+```
+
+**Assuming all utilized Ops and Op features are still available**, the Op can then be reconstructed using the method `OpEnvironment.opFromSignature()`:
+
+```java
+// The value of signature could be saved within a script for future reconstruction
+String signature = Ops.signature(op);
+
+...
+
+// NB This generic type construct is needed for the OpEnvironment to provide a Computer Op.
+var opType = new Nil<Computers.Arity2<Img<DoubleType>, Double, Img<DoubleType>>>() {};
+// Reconstruct the Op
+var sameOp = ops.opFromSignature(signature, opType);
+// We could then call this Op with our input image, real-valued sigma, and output image container
+sameOp.compute(img, sigma, out);
+```
+
+There still exist limitations with reconstructive features, including:
+* [scijava/scijava#220](https://github.com/scijava/scijava/issues/220) - The inability to reconstruct an Op using simpler `opType`s.
+* [scijava/scijava#219](https://github.com/scijava/scijava/issues/219) - The inability to reconstruct Op matches with omitted optional parameters.
+
+## Info Trees
+
+Ops often utilize Op dependencies, which may have dependencies themselves. As such, the Ops returned by SciJava Ops are best thought of as a [tree](https://en.wikipedia.org/wiki/Tree_(data_structure)). While an Op's signature, described above, describes the entire Tree, it is often more convenient to interrogate dependencies using the `InfoTree` data structure, obtained using the method `Ops.infoTree()`:
+
+```java
+var tree = Ops.infoTree(op);
+```
+
+The `InfoTree` API allows users to interrogate:
+* The root `OpInfo`, accessible using `InfoTree.info()`
+* All dependencies, which are each themselves an `InfoTree`, using `InfoTree.dependencies()`.
+
+## Op History
+
+It is often useful to determine the list of Ops responsible for the current state of a given Object. To aid in this task, each `OpEnvironment` records Op executions within an `OpHistory` object, accessible using the method `OpEnvironment.history()`.
+
+```java
+// OpEnvironment.history() provides access to a OpHistory object
+var history = ops.history();
+
+// Since an Op was responsible for the results in out, it is listed in OpHistory.executionsUpon()
+var usedOps = history.executionsUpon(out);
+```
+
+Using the `OpHistory` object, we can see that our gaussian blur Op `op` is at the end of the list, since it mutated `out` last. If `out` was additionally created using SciJava Ops, the Op(s) responsible for creating the image would also be included.
```

Comment:
```suggestion
Using the `OpHistory` object, we can see that our gaussian blur Op `op` is at the end of the list, since it wrote to `out` last. If `out` was additionally created using SciJava Ops, the Op(s) responsible for creating the image would also be included.
```